CharlieBackend.Core/ResultExtensionMethods.cs

```diff
@@ -10,32 +10,42 @@ public static class ResultExtensionMethods
         /// <param name="result">Result T variable with data or error to transfer to consumer from controller</param>
-        /// <returns>Status code depending Result T data. If ErrorData is not empty, return error status code.
+        /// <returns>Status code depending Result T data. If ErrorData is not empty, return error status code,
+        /// and message in Json format
         /// If no error transferred, returns OkResult (200 or 204 status code)</returns>
         public static ActionResult ToActionResult<T>(this Result<T> result)
         {
             if (Equals(result, null))
             {
-                return new ObjectResult("No data received while processing data model") { StatusCode = 500 };
+                result = new Result<T>
+                {
+                    Error = new ErrorData()
+                    {
+                        Code = ErrorCode.InternalServerError,
+                        Message = "No data received while processing data model"
+                    }
+                };
+
+                return new JsonResult(ConverToJson(result)) { StatusCode = 500 };
             }
-            else if (result.ErrorData != null)
+            else if (result.Error != null)
             {
-                switch (result.ErrorData.ErrorCode)
+                switch (result.Error.Code)
                 {
                     case ErrorCode.Unauthorized:
-                        return new UnauthorizedObjectResult(result.ErrorData.ErrorMessage);//401
+                        return new JsonResult(ConverToJson(result)) { StatusCode = 401 };//401
                     case ErrorCode.ValidationError:
-                        return new BadRequestObjectResult(result.ErrorData.ErrorMessage);//400
+                        return new JsonResult(ConverToJson(result)) { StatusCode = 400 };//400
                     case ErrorCode.InternalServerError:
-                        return new StatusCodeResult(500);//500
+                        return new JsonResult(ConverToJson(result)) { StatusCode = 500 };//500
                     case ErrorCode.NotFound:
-                        return new NotFoundObjectResult(result.ErrorData.ErrorMessage);//404
+                        return new JsonResult(ConverToJson(result)) { StatusCode = 404 };//404
                     case ErrorCode.UnprocessableEntity:
-                        return new UnprocessableEntityObjectResult(result.ErrorData.ErrorMessage);//422
+                        return new JsonResult(ConverToJson(result)) { StatusCode = 422 };//422
                     case ErrorCode.Conflict:
-                        return new ConflictObjectResult(result.ErrorData.ErrorMessage);//409
+                        return new JsonResult(ConverToJson(result)) { StatusCode = 409 };//409
                     default:
-                        return new StatusCodeResult(500);
+                        return new JsonResult(ConverToJson(result)) { StatusCode = 500 };
```

Comment:
you don't need ConverToJson method, you can pass result.Error to JsonResult

Reply:
Ok.

Made it due to REST recommendations for error response (to have "error" parent class)

Reply:
We do not need to serialize object to Json, before passing it to JsonResult, that was main message of my comment. The name that you choose was confisung, because actually what ConverToJson does is simply mapping one object to another.

I agree that it would be better to have error as property of a json object. Let's create an ErrorDto, which will contain ErrorData Error { get; set; }. In that way, we wouldn't output empty Data property, and Error will be a property of an object.